flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamSort.scala

```diff
@@ -0,0 +1,167 @@
+
+  override def deriveRowType(): RelDataType = rowRelDataType
+
+  override def copy(traitSet: RelTraitSet, inputs: java.util.List[RelNode]): RelNode = {
+    new DataStreamSort(
+      calc,
+      cluster,
+      traitSet,
+      inputs.get(0),
+      rowRelDataType,
+      inputType,
+      description + calc.getId())
+  }
+
+  override def toString: String = {
+    s"Sort($calc)"+
+      s"on fields: (${calc.collation.getFieldCollations})"
+  }
+
+  override def explainTerms(pw: RelWriter): RelWriter = {
+    super.explainTerms(pw)
+      .item("aggregate", calc)
+      .item("sort fields",calc.collation.getFieldCollations)
+      .itemIf("offset", calc.offset, calc.offset!=null)
+      .itemIf("fetch", calc.fetch, calc.fetch!=null)
+      .item("input", inputNode)
+  }
+
+  override def translateToPlan(tableEnv: StreamTableEnvironment): DataStream[Row] = {
+
+    val inputDS = getInput.asInstanceOf[DataStreamRel].translateToPlan(tableEnv)
+    
+    //need to identify time between others order fields. Time needs to be first sort element
+    val timeType = SortUtil.getTimeType(calc,inputType)
+    
+    //time ordering needs to be ascending
+    if(SortUtil.getTimeDirection(calc)!=Direction.ASCENDING) {
+      throw new TableException("SQL/Table supports only ascending time ordering")
+    }
+      
+     
+    val (offset,fetch) = (calc.offset,calc.fetch)
+    
+    //enable to extend for other types of aggregates that will not be implemented in a window
+    timeType match {
+        case _: ProcTimeType =>
+            (offset,fetch) match {
+              case (o:Any,f:Any)  => null             // offset and fetch needs retraction
+              case (_,f:Any) => null                  // offset needs retraction
+              case (o:Any,_) => null                  // fetch needs retraction
+              case _ => createSortProcTime(inputDS)   //sort can be done with/without retraction
+            }
+        case _: RowTimeType =>
+          throw new TableException("SQL/Table does not support sort on row time")
+        case _ =>
+          throw new TableException("SQL/Table needs to have sort on time as first sort element")
+    }
+    
+  }
+
+  /**
+   * Create Sort logic based on processing time
+   */
+  def createSortProcTime(
+    inputDS: DataStream[Row]): DataStream[Row] = {
+
+    /*
+     * [TODO] consider the partition case - In this implementation even if the stream is
+     * pre-partitioned all elements will be reshufled and sorted
+     */
+
+        
+     // get the output types
+    //Sort does not do project.= Hence it will output also the ordering proctime field
+    //[TODO]Do we need to drop the ordering fields?
+    val rowTypeInfo = FlinkTypeFactory.toInternalRowTypeInfo(getRowType).asInstanceOf[RowTypeInfo]
+    
+    val processFunction = SortUtil.createProcTimeSortFunction(calc,inputType)
+      
+     val result: DataStream[Row] = inputDS
+            .keyBy(new NullByteKeySelector[Row])
```

Comment:
@fhueske 
Having a null partition for the process function can break previous partitions perhaps and reshuffle all elements again into one function. Perhaps we need to drop this - what do you think?
Also...we need to check that even without partition functions we can still use ValueState (i think it is possible as i believe the MapState had a pre-requisite to have partitioned functions) is it so?